src/fiber.cr

```diff
@@ -86,11 +89,18 @@ class Fiber
   # :nodoc:
   def initialize
     @proc = Proc(Void).new { }
-    @stack = Pointer(Void).null
     @stack_top = _fiber_get_stack_top
     @stack_bottom = GC.stack_bottom
     @name = "main"
 
+    @stack = Pointer(Void).null
+    {% unless flag?(:win32) %}
+      if LibC.getrlimit(LibC::RLIMIT_STACK, out rlim) == 0
+        stack_size = rlim.rlim_cur
+        @stack = Pointer(Void).new(@stack_bottom.address - stack_size)
+      end
+    {% end %}
```

Comment:
That should work for the main thread, but is it correct for created threads? I don't know what's the default stack size for them. Maybe we should use [pthread_getattr_np(3)](http://man7.org/linux/man-pages/man3/pthread_getattr_np.3.html) and [pthread_attr_getstacksize(3)](http://man7.org/linux/man-pages/man3/pthread_attr_getstacksize.3.html) for created thread?

Reply:
Let's not worry about that for now until after threaded GC works. Otherwise it's stalling the PR.

Reply:
Sure, but maybe we can add a comment about it :)

Reply:
Yeah, a `TODO: ` would be nice